lib/reverseproxy/reverse_proxy.go

```diff
@@ -0,0 +1,130 @@
+package reverseproxy
+
+import (
+	"flag"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/bytesutil"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+)
+
+const (
+	targetURL     = "vm-target-url"
+	forwardHeader = "X-Forwarded-For"
+)
+
+var (
+	maxIdleConnsPerBackend = flag.Int("maxIdleConnsPerBackend", 100, "The maximum number of idle connections vmauth can open per each backend host")
+)
+
+var bbPool bytesutil.ByteBufferPool
+
+// ReverseProxy represents simple revers proxy based on http.Client
+type ReverseProxy struct {
+	client *http.Client
+}
+
+// New initialize revers proxy
+func New() *ReverseProxy {
+	tr := prepareTransport()
+	return &ReverseProxy{
+		client: &http.Client{
+			Transport: tr,
```

Comment:
no need to have separate variable for transport (tr:=)

Reply:
fixed